xl-v1.go

```diff
@@ -94,6 +136,14 @@ func newXL(disks ...string) (StorageAPI, error) {
 	// Save all the initialized storage disks.
 	xl.storageDisks = storageDisks
 
+	xl.nameSpaceLockMap = make(map[nameSpaceParam]nameSpaceLock)
+	xl.nameSpaceLockMapMutex = &sync.Mutex{}
+	xl.readQuorum = len(xl.storageDisks) / 2
+	xl.writeQuorum = xl.readQuorum + 3
+	if xl.writeQuorum > len(xl.storageDisks) {
+		xl.writeQuorum = len(xl.storageDisks)
+	}
```

Comment:
> this means that for 4 or 6 disks we will not write even if one disk is down. needs to be discussed.

This needs to be fixed - readQuorum itself requires N/2 + 1 disks. Write quorum is also N/2 + 1, we will atleast write 1 parity and atleast have 1 parity to read. 

This is needed since the matrix type we are using called 'vandermonde' implemented in klauspost requires you to make sure that you have N/2 + 1 for consistent reconstruction.